### PR TITLE
[vcpkg_fail_port_install] add a warning about extra args when ALWAYS is passed

### PR DESCRIPTION
**Pull request by strega-nil**
cc @horenmar

### CODE REVIEW

File: scripts/cmake/vcpkg_fail_port_install.cmake [hunk 1]
```diff
@@ -40,23 +40,42 @@ Library linkage for which the build should fail early.
...
+
+    if(arg_ALWAYS)
+        vcpkg_list(SET extra_args)
+        foreach(arg IN LISTS multi_args)
+            if(DEFINED "arg_${arg}")
+                vcpkg_list(APPEND extra_args "${arg}" "${arg_${arg}}")
+            endif()
+        endif()
+        if(NOT "${extra_args}" STREQUAL "")
+            message(WARNING "vcpkg_fail_port_install(ALWAYS) was passed extra arguments: ${extra_args}")
```
Comment (horenmar) [inline scripts/cmake/vcpkg_fail_port_install.cmake:57]: I feel like this doesn't explain the reason for the warning too well. How about something like

```suggestion
            message(WARNING "vcpkg_fail_port_install set to fail both unconditionally and conditionally on ${extra_args}. This is likely to be an error.")
```

**Review by BillyONeal — APPROVED** [review]

File: scripts/cmake/vcpkg_fail_port_install.cmake [hunk 2]
```diff
@@ -40,23 +40,42 @@ Library linkage for which the build should fail early.
...
         message(FATAL_ERROR "vcpkg_fail_port_install was passed extra arguments: ${arg_UNPARSED_ARGUMENTS}")
     endif()
+
+    if(arg_ALWAYS)
+        vcpkg_list(SET extra_args)
+        foreach(arg IN LISTS multi_args)
+            if(DEFINED "arg_${arg}")
+                vcpkg_list(APPEND extra_args "${arg}" "${arg_${arg}}")
+            endif()
+        endif()
```
Comment (PhoebeHui) [inline scripts/cmake/vcpkg_fail_port_install.cmake:55]:
```suggestion
        endforeach()
```

**Review by PhoebeHui — APPROVED** [review]

**Comment by PhoebeHui** [comment]
Depends on https://github.com/microsoft/vcpkg/pull/21825

**Comment by PhoebeHui** [comment]
tensorflow:x64-osx and tensorflow-cc:x64-osx failed due to https://github.com/microsoft/vcpkg/issues/21758.

**Comment by PhoebeHui** [comment]
I can't repro chromium-base:x64-osx libcanberra:x64-linux failures locally, mpg123:x64-linux has been fixed in master.

chromium-base:x64-osx
```
FAILED: obj/base/base/superfasthash.o 
../../../../../../usr/bin/clang -MMD -MF obj/base/base/superfasthash.o.d -DSYSTEM_NATIVE_UTF8 -D_LIBCPP_HAS_NO_ALIGNED_ALLOCATION -DCR_XCODE_VERSION= -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__ASSERT_MACROS_DEFINE_VERSIONS_WITHOUT_UNDERSCORES=0 -D_DEBUG -DDYNAMIC_ANNOTATIONS_ENABLED=1 -DBASE_IMPLEMENTATION -I../src/25ce732 -Igen -fno-strict-aliasing -fstack-protector-strong -fcolor-diagnostics -fmerge-all-constants -fcrash-diagnostics-dir=../src/25ce732/tools/clang/crashreports -Xclang -mllvm -Xclang -instcombine-lower-dbg-declare=0 -arch x86_64 -Wno-builtin-macro-redefined -D__DATE__= -D__TIME__= -D__TIMESTAMP__= -no-canonical-prefixes -Wall -Wextra -Wimplicit-fallthrough -Wunreachable-code -Wthread-safety -Wextra-semi -Wunguarded-availability -Wno-missing-field-initializers -Wno-unused-parameter -Wno-c++11-narrowing -Wno-unneeded-internal-declaration -Wno-undefined-var-template -Wno-ignored-pragma-optimize -Wno-implicit-int-float-conversion -Wno-final-dtor-non-final-class -Wno-builtin-assume-aligned-alignment -Wno-deprecated-copy -Wno-non-c-typedef-for-linkage -Wmax-tokens -O0 -fno-omit-frame-pointer -fno-standalone-debug -gdwarf-4 -g2 -isysroot ../../../../../../Library/Developer/CommandLineTools/SDKs/MacOSX10.15.sdk -mmacosx-version-min=10.10.0 -ftrivial-auto-var-init=pattern -fvisibility=hidden -Wheader-hygiene -Wstring-conversion -Wtautological-overlap-compare -Wno-unknown-warning-option -Wno-unknown-pragmas -Wglobal-constructors -Wexit-time-destructors -Wshadow -Wexit-time-destructors -std=c11 -Wno-implicit-fallthrough -include obj/base/base/precompile.h-c -c ../src/25ce732/base/third_party/superfasthash/superfasthash.c -o obj/base/base/superfasthash.o
/bin/sh: ../../../../../../usr/bin/clang: No such file or directory
```

**Comment by BillyONeal** [comment]
In new port code I think we should suggest just using `message(FATAL_ERROR` directly rather than calling this with `ALWAYS`.

**Review by BillyONeal — APPROVED** [review]
This is OK because it's adding a diagnostic for a problematic problem but I think the correct fix is just to deprecate `ALWAYS` outright in favor of `message(FATAL_ERROR`.

**Comment by BillyONeal** [comment]
(Dropping info:reviewed because @strega-nil can merge if she wants)

**Comment by strega-nil-ms** [comment]
Thanks Billy!

**Pull request merged**